plugins/org.eclipse.xtext/src/org/eclipse/xtext/generator/URIBasedFileSystemAccess.xtend

```diff
@@ -73,9 +77,10 @@ class URIBasedFileSystemAccess extends AbstractFileSystemAccess2 {
 	
 	override generateFile(String fileName, String outputCfgName, CharSequence contents) {
 		val uri = getURI(fileName, outputCfgName)
-		generateTrace(fileName, outputCfgName, contents)
 		val encoding = getEncoding(uri)
```

Comment:
Hmm I wonder how this could work, e.g. if your source lang is XML you have the XMLEncodingProvider. That one would not be able to provide encoding for other files in general (and vice versa). Also other langs shouldn't care that some file type has a fixed encoding. For example, properties files should usually be ISO-8859-1 whereas most other languages are likely to be UTF-8.
Also we do this already - JavaIoFileSystemAccess.getEncoding(URI)) will only fall back to the source languages encoding provider if the target language is not known the Xtext.